rootfs/etc/nginx/lua/balancer/hashcode.lua

```diff
@@ -0,0 +1,17 @@
+local string = string
+local _M = {}
+
+function _M.str_hash_to_int(str)
+    local h = 0
+    local l = #str
+    if l > 0 then
+        local i = 0
+        while i < l do
+            h = 31 * h + string.byte(str, i + 1);
```

Comment:
Where does 31 come from? Can you explain the magic here a bit?

Reply:
the hashcode method came from java string hashcode
1. 31 is a prime number which can decrease hash collision attack
2. that can be optimized ,  31 * i = (i << 5) - i